ehcache-impl/src/main/java/org/ehcache/impl/config/SizedResourcePoolImpl.java

```diff
@@ -46,17 +46,31 @@ public class SizedResourcePoolImpl<P extends SizedResourcePool> extends Abstract
    * @param persistent whether the pool is to be persistent
    */
   public SizedResourcePoolImpl(ResourceType<P> type, long size, ResourceUnit unit, boolean persistent) {
-    super(type, persistent);
+    this(type, size, unit, persistent, false);
+  }
+
+  /**
+   * Creates a new resource pool based on the provided parameters.
+   *
+   * @param type the resource type
+   * @param size the size
+   * @param unit the unit for the size
+   * @param persistent whether the pool is to be persistent
+   * @param shared whether the pool is to be shared
+   */
+  public SizedResourcePoolImpl(ResourceType<P> type, long size, ResourceUnit unit, boolean persistent, boolean shared) {
+    super(type, persistent, shared);
     if (unit == null) {
       throw new NullPointerException("ResourceUnit can not be null");
     }
-    if (size <= 0) {
+    if (size <= 0 && !shared) {
       throw new IllegalArgumentException("Size must be greater than 0");
```

Comment:
A SizedResourcePool that is marked for sharing and has zero bytes is used to inform the cache manager that the cache wants to use a shared heap or offheap sized pool.